src/components/widgets/NumberAndDateInputs.tsx

```diff
@@ -92,10 +95,11 @@ function BaseInput({
   const boundsCheckedValue = useCallback(
     (newValue?: NumberOrDate) => {
       if (newValue == null) {
-        setErrorState({
-          error: true,
-          helperText: `Please enter a ${valueType}.`,
-        });
+        required &&
+          setErrorState({
+            error: true,
+            helperText: `Please enter a ${valueType}.`,
+          });
         return false;
```

Comment:
I think we should notify if `required` is `false`. Presumably, in this case, `undefined` is a valid value for the consumer.

```suggestion
        return !required;
```

Reply:
Wait, I just realized I was reviewing the wrong feature! This should have been added to https://github.com/VEuPathDB/web-components/pull/99. I will resolve this conversation and open a new issue. Sorry!

Reply:
Ah, wait again! This change was made in this PR. I will unresolve this :smile: